gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/tpgm/snapshot/Snapshot.java

```diff
@@ -53,26 +50,15 @@ public Snapshot(TemporalPredicate predicate) {
 
   @Override
   public TemporalGraph execute(TemporalGraph superGraph) {
-    DataSet<TempVertexTuple> vertices = superGraph.getVertices()
-      .map(new TemporalVertexToTempVertexTuple())
-      .filter(new ByTemporalPredicate<>(temporalPredicate));
-    DataSet<TempEdgeTuple> edges = superGraph.getEdges().map(new TemporalEdgeToTempEdgeTuple())
-      .filter(new ByTemporalPredicate<>(temporalPredicate));
-
-    DataSet<TempEdgeTuple> verifiedEdges = edges
-      .join(vertices).where(1).equalTo(0).with(new LeftSide<>())
-      .join(vertices).where(2).equalTo(0).with(new LeftSide<>());
-
-    DataSet<TemporalVertex> originalVertices = superGraph.getVertices();
-    DataSet<TemporalEdge> originalEdges = superGraph.getEdges();
+    DataSet<TemporalVertex> vertices = superGraph.getVertices()
+      // Filter vertices
+      .filter(new ByTemporalPredicate<>(temporalPredicate))
+      .name("Snapshot vertices by [" + temporalPredicate.toString() + "]");
```

Comment:
Same thing here, you can remove the explicit `toString` calls or remove the call to `name` by adding a `toString` implementation to `ByTemporalPredicate`

Reply:
Have you tested it?